osu.Game/Scoring/ScoreManager.cs

```diff
@@ -66,12 +74,41 @@ protected override ScoreInfo CreateModel(ArchiveReader archive)
                 }
                 catch (LegacyScoreDecoder.BeatmapNotFoundException e)
                 {
-                    Logger.Log(e.Message, LoggingTarget.Information, LogLevel.Error);
+                    Logger.Log(e.Message, LoggingTarget.Information, LogLevel.Error, false);
+                    sendNotification(e, archive);
                     return null;
                 }
             }
         }
 
+        private void sendNotification(LegacyScoreDecoder.BeatmapNotFoundException e, ArchiveReader archive)
+        {
+            var req = new GetBeatmapRequest(e.BeatmapHash);
+
+            var notification = new SimpleErrorNotification
+            {
+                Text = e.Message
+            };
+
+            req.Success += res =>
+            {
+                notification.Text += " Click here to download it.";
```

Comment:
So am I to understand that the notification text is potentially going to change *while the user is reading it*, without any indication? That seems sub-optimal. (Also, this will be unlocalisable at present.)

Needs to be re-thought. Maybe the clicking of the notification should fire a request to try to find the beatmap, and show another notification (replacing the previous one) that the lookup failed. Or the "replay import" screen could show an error message. Not sure, needs experimentation.

Reply:
> So am I to understand that the notification text is potentially going to change _while the user is reading it_, without any indication? That seems sub-optimal. (Also, this will be unlocalisable at present.)
> 
> Needs to be re-thought. Maybe the clicking of the notification should fire a request to try to find the beatmap, and show another notification (replacing the previous one) that the lookup failed. Or the "replay import" screen could show an error message. Not sure, needs experimentation.

I don't think so? The notification posts after the request fails or success

Reply:
Ok, I misread, but localisation-wise the ad-hoc concatenation is still pretty bad.